### PR TITLE
feature: reset/unstage index entries

### PR DESCRIPTION
**Pull request by antis81**
This one brings in the ability to unstage files from the index.

### CODE REVIEW

File: libGitWrap/GitWrap.cpp [hunk 1]
```diff
@@ -46,6 +46,42 @@
...
+        }
+
+        StrArrayWrapper::StrArrayWrapper(const QStringList& sl)
+        {
+            internalCopy = sl;
+
+            a.count = sl.count();
+            a.strings = new char *[a.count];
+
+            for(int i=0; i < sl.count(); ++i )
```
Comment (scunz) [inline libGitWrap/GitWrap.cpp]: should use internalCopy, too.

**Comment by antis81** [comment]
Think we can merge this as well, or?

**Comment by scunz** [comment]
Isn't the StrArrayWrapper (introduced here) used by the PR you just merged? I'm a bit confused. But anyway, this PR doesn't look bad either. So I'd say: go ahead.

**Comment by antis81** [comment]
Yes. The `checkout` PR goes on this branch and not on `development`. It would have been better to keep that StrArray thingy in another `c2cpp_internal` branch (and PR respectively), that contains internal core functionality to multiple API methods ... Anyways, next time I'll do better :innocent:.

So I'm going to merge here and the `WorkingTree` feature branch and refactoring branches (which will need some cleanup in this branch for the "new" functions) in turn.

**Pull request merged**